packages/datagateway-common/src/handleICATError.ts

```diff
@@ -0,0 +1,40 @@
+import { AxiosError } from 'axios';
+import * as log from 'loglevel';
+
+const handleICATError = (
+  error: AxiosError,
+  broadcast: boolean = true
+): void => {
+  log.error(error.message);
+  if (broadcast) {
+    document.dispatchEvent(
+      new CustomEvent('scigateway', {
+        detail: {
+          type: 'scigateway:api:notification',
+          payload: {
+            severity: 'error',
+            message: error.message,
+          },
+        },
+      })
+    );
+  }
+  if (
+    error.response &&
```

Comment:
Currently the token is being invalidated on specific 403 messages which makes sense, I wonder if it should be broadened to invalidate on general error messages in case someone is trying to get in and manages to break the system in a new way.

Reply:
We don't want to invalidate token on general errors because what if the user gets a server error - should that log them out? Limiting it to only invalidate the token on specific, invalid session id related errors means the user is prompted to re-login only when they have to.

Reply:
understood!